<launch>
  <!-- Camera topic -->
  <arg name="topic"           default="/camera/image_rect_color" />

  <!-- Frames for table and camera -->
  <arg name="table_frame"     default="/table_frame" />
  <arg name="camera_frame"    default="/camera_frame" />

  <!-- Camera topic -->
  <arg name="data_path"       default="$(find saphari_tool_detector)/data/" />

  <!-- Threshold for edge detection -->
  <arg name="threshold_low"   default="50.0" />
  <arg name="threshold_high"  default="100.0" />

  <!-- Threshold for hough tool detection -->
  <arg name="threshold_hough" default="50" />

  <!-- Maximum allowed overlap -->
  <arg name="max_overlap"     default="0.4" />

  <!-- Bounding box for tool detection area -->
  <arg name="x"               default="200" />
  <arg name="y"               default="0" />
  <arg name="width"           default="1600" />
  <arg name="height"          default="1199" />

  <node pkg="saphari_tool_detector" type="detect_tools" name="tool_detector" ns="tool_detector" >
    <param name="topic"           type="str"    value="$(arg topic)" />
    <param name="table_frame"     type="str"    value="$(arg table_frame)" />
    <param name="camera_frame"    type="str"    value="$(arg camera_frame)" />
    <param name="data_path"       type="str"    value="$(arg data_path)" />
    <param name="threshold_low"   type="double" value="$(arg threshold_low)" />
    <param name="threshold_high"  type="double" value="$(arg threshold_high)" />
    <param name="threshold_hough" type="int"    value="$(arg threshold_hough)" />
    <param name="max_overlap"     type="double" value="$(arg max_overlap)" />
    <param name="x"               type="int"    value="$(arg x)" />
    <param name="y"               type="int"    value="$(arg y)" />
    <param name="width"           type="int"    value="$(arg width)" />
    <param name="height"          type="int"    value="$(arg height)" />
  </node>
</launch>
